### PR TITLE
Modernize example and show that it throws

### PR DESCRIPTION
**Pull request by marcoscaceres**
***
<a href="https://pr-preview.s3.amazonaws.com/WICG/web-share/pull/56.html" title="Last updated on Feb 12, 2018, 3:05 AM GMT (16f99af)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/web-share/56/d39a1a7...16f99af.html" title="Last updated on Feb 12, 2018, 3:05 AM GMT (16f99af)">Diff</a>

### CODE REVIEW

**Comment by marcoscaceres** [comment]
Single quotes are the devil :)

**Comment by maxnordlund** [comment]
Does it really throw? Seems like it returns a promise, and only because your example uses async/await it turns it into a throw, right?

Anyway, having both a promise based, and async/await based example is nice.

File: index.html [hunk 1]
```diff
@@ -60,9 +60,13 @@ <h2>
...
       <pre class="example javascript" title="Basic usage">
-        shareButton.addEventListener('click', () =&gt; {
-          navigator.share({title: 'Example Page', url: ''})
-              .then(console.log('Share successful'));
+        shareButton.addEventListener('click', async () =&gt; {
+          try {
+            await navigator.share({title: 'Example Page', url: ''});
+            console.log('Share successful');
+          } catch (err) {
+            console.error('Oh no! couldn't share. ' + err.message);
```
Comment (ericwilligers) [inline index.html]: The single quote in `couldn't` needs escaping.

**Comment by marcoscaceres** [comment]
> Does it really throw? Seems like it returns a promise, and only because your example uses async/await it turns it into a throw, right?

Correct.

> Anyway, having both a promise based, and async/await based example is nice.

Given async/await, using raw promise (particularly examples that use `.then()` and `.catch()`) is a bit of an anti-pattern - so I'd like us not to just stick to async/await examples. Keeps things much cleaner. We should only revert back to `.then`, `.catch()`, etc. if there is a valid reason to do so.

File: index.html [hunk 2]
```diff
@@ -60,9 +60,13 @@ <h2>
...
       <pre class="example javascript" title="Basic usage">
-        shareButton.addEventListener('click', () =&gt; {
-          navigator.share({title: 'Example Page', url: ''})
-              .then(console.log('Share successful'));
+        shareButton.addEventListener("click", async () => {
+          try {
+            await navigator.share({ title: "Example Page", url: "" });
+            console.log("Share successful");
+          } catch (err) {
+            console.error("Oh no! couldn't share.", err.message);
```
Comment (mgiuca) [inline index.html]: I don't like this informal message. Can you change the string to "Share failed:"

**Pull request merged**